### PR TITLE
Add custom result index lifecycle management condition fields to config

### PR DESCRIPTION
**Pull request by jackiehanyang**
### Description
Add custom result index lifecycle management condition fields to config

### Issues Resolved
https://github.com/opensearch-project/anomaly-detection/issues/1213

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/timeseries/model/Config.java [hunk 1]
```diff
@@ -294,6 +306,21 @@ public Config(StreamInput input) throws IOException {
         this.recencyEmphasis = input.readInt();
         this.seasonIntervals = input.readInt();
         this.historyIntervals = input.readInt();
+        if (input.readBoolean()) {
+            this.customResultIndexMinSize = input.readInt();
+        } else {
+            customResultIndexMinSize = null;
```
Comment (kaituo) [inline src/main/java/org/opensearch/timeseries/model/Config.java:312]: can you add this. for all 3 fields to make the style consistent?

File: src/main/java/org/opensearch/timeseries/model/Config.java [hunk 2]
```diff
@@ -343,6 +370,21 @@ public void writeTo(StreamOutput output) throws IOException {
         output.writeInt(recencyEmphasis);
         output.writeInt(seasonIntervals);
         output.writeInt(historyIntervals);
+        if (customResultIndexMinSize != null) {
```
Comment (kaituo) [inline src/main/java/org/opensearch/timeseries/model/Config.java:373]: Can you just to use writeOptionalInt and readOptionalInt that would writeBoolean true/false based on null check?

File: src/main/java/org/opensearch/ad/model/AnomalyDetector.java [hunk 3]
```diff
@@ -330,6 +348,21 @@ public void writeTo(StreamOutput output) throws IOException {
         } else {
             output.writeBoolean(false);
         }
+        if (customResultIndexMinSize != null) {
+            output.writeOptionalInt(customResultIndexMinSize);
```
Comment (kaituo) [inline src/main/java/org/opensearch/ad/model/AnomalyDetector.java:352]: You can remove null check as writeOptinalInt covers that.

File: src/main/java/org/opensearch/timeseries/model/Config.java (hunk 2 above)
Comment (jackiehanyang) [reply src/main/java/org/opensearch/timeseries/model/Config.java:373]: sure, will update it

File: src/main/java/org/opensearch/ad/model/AnomalyDetector.java (hunk 3 above)
Comment (jackiehanyang) [reply src/main/java/org/opensearch/ad/model/AnomalyDetector.java]: using `writeOptinalInt` in both AD and Config file

File: src/main/java/org/opensearch/ad/model/AnomalyDetector.java [hunk 4]
```diff
@@ -268,14 +277,23 @@ public AnomalyDetector(StreamInput input) throws IOException {
         if (input.readBoolean()) {
             this.rules = input.readList(Rule::new);
         }
+        if (input.readBoolean()) {
```
Comment (kaituo) [inline src/main/java/org/opensearch/ad/model/AnomalyDetector.java:280]: You can use readOptionalInt now.

File: src/main/java/org/opensearch/timeseries/model/Config.java [hunk 5]
```diff
@@ -294,6 +306,21 @@ public Config(StreamInput input) throws IOException {
         this.recencyEmphasis = input.readInt();
         this.seasonIntervals = input.readInt();
         this.historyIntervals = input.readInt();
+        if (input.readBoolean()) {
```
Comment (kaituo) [inline src/main/java/org/opensearch/timeseries/model/Config.java:309]: You can use readOptionalInt now.

File: src/main/java/org/opensearch/ad/model/AnomalyDetector.java (hunk 4 above)
Comment (jackiehanyang) [reply src/main/java/org/opensearch/ad/model/AnomalyDetector.java]: good catch! forgot to update this part

File: src/main/java/org/opensearch/timeseries/model/Config.java (hunk 5 above)
Comment (jackiehanyang) [reply src/main/java/org/opensearch/timeseries/model/Config.java]: Updated

**Review by kaituo — APPROVED** [review]

**Review by kaituo — APPROVED** [review]

**Pull request merged**